File: app/src/main/java/com/simon/harmonichackernews/ui/common/CaptchaResultCallback.java
package com.simon.harmonichackernews.ui.common;

import androidx.annotation.NonNull;

import com.simon.harmonichackernews.network.UserActions;

/** Callback bridge for Compose-owned CAPTCHA dialogs requested by non-UI coordinators. */
public interface CaptchaResultCallback {
    void onCaptchaResponse(@NonNull UserActions.CaptchaChallenge challenge,
                           @NonNull String captchaResponse);

    void onCaptchaCancelled();
}

File: app/src/main/java/com/simon/harmonichackernews/ui/common/CaptchaResultCallback.kt
package com.simon.harmonichackernews.ui.common

import com.simon.harmonichackernews.network.HackerNewsCaptchaChallenge

/** Callback bridge for Compose-owned CAPTCHA dialogs requested by non-UI coordinators.  */
interface CaptchaResultCallback {
    fun onCaptchaResponse(
        challenge: HackerNewsCaptchaChallenge,
        captchaResponse: String,
    )

    fun onCaptchaCancelled()
}
